v4/newrelic/application.go

```diff
@@ -55,7 +66,13 @@ func (app *Application) StartTransaction(name string) *Transaction {
 // https://docs.newrelic.com/docs/insights/new-relic-insights/adding-querying-data/inserting-custom-events-new-relic-apm-agents
 //
 // An error is logged if eventType or params is invalid.
-func (app *Application) RecordCustomEvent(eventType string, params map[string]interface{}) {}
+func (app *Application) RecordCustomEvent(eventType string, params map[string]interface{}) {
+	if app == nil {
+		return
+	}
+	app.initLogger()
```

Comment:
Why is `app.initLogger()` being called here when it isn't being called before `logUnimplemented` in `app.Shutdown`?